### PR TITLE
Asssume app is terminated when the port is closed.

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD https://ocd.appscale.com:8080/job/Daily%20Build/1694/

File: appscale/tools/appscale_tools.py [hunk 1]
```diff
@@ -363,14 +363,37 @@ def remove_app(cls, options):
     if not acc.does_app_exist(options.appname):
       raise AppScaleException("The given application is not currently running.")
 
+    # Makes a call to the AppController to get all the stats and looks
+    # through them for the http port the app can be reached on.
+    http_port = None
+    for i in range(cls.MAX_RETRIES):
+      try:
```
Comment (cdonati) [inline appscale/tools/appscale_tools.py]: Put the try/except logic only around the lines that need it.

File: appscale/tools/appscale_tools.py [hunk 2]
```diff
@@ -363,14 +363,37 @@ def remove_app(cls, options):
...
+
     acc.stop_app(options.appname)
     AppScaleLogger.log("Please wait for your app to shut down.")
-    while True:
-      if acc.is_app_running(options.appname):
+
+    for i in range(cls.MAX_RETRIES):
+      if RemoteHelper.is_port_open(login_host, http_port, options.verbose):
         time.sleep(cls.SLEEP_TIME)
+        AppScaleLogger.success("Waiting for {0} to terminate...".format(options.appname))
```
Comment (cdonati) [inline appscale/tools/appscale_tools.py]: Keep the line length under 80 characters.

File: appscale/tools/appscale_tools.py [hunk 3]
```diff
@@ -363,14 +363,37 @@ def remove_app(cls, options):
     if not acc.does_app_exist(options.appname):
       raise AppScaleException("The given application is not currently running.")
 
+    # Makes a call to the AppController to get all the stats and looks
+    # through them for the http port the app can be reached on.
+    http_port = None
+    for i in range(cls.MAX_RETRIES):
```
Comment (cdonati) [inline appscale/tools/appscale_tools.py]: pylint dislikes single-letter variable names.

File: appscale/tools/appscale_tools.py (hunk 2 above)
Comment (menivaitsi) [inline appscale/tools/appscale_tools.py]: Can this be a simple log instead of succes?

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1707/

File: appscale/tools/appscale_tools.py [hunk 4]
```diff
@@ -360,17 +360,44 @@ def remove_app(cls, options):
...
     acc = AppControllerClient(login_host, secret)
 
-    if not acc.does_app_exist(options.appname):
+    if not acc.is_app_running(options.appname):
       raise AppScaleException("The given application is not currently running.")
 
+    # Makes a call to the AppController to get all the stats and looks
+    # through them for the http port the app can be reached on.
+    http_port = None
+    for _ in range(cls.MAX_RETRIES):
```
Comment (cdonati) [inline appscale/tools/appscale_tools.py]: Minor nitpick: `MAX_RETRIES` is being used as total tries here.

File: appscale/tools/appscale_tools.py [hunk 5]
```diff
@@ -360,17 +360,44 @@ def remove_app(cls, options):
...
+    if not http_port:
+      raise AppScaleException(
+        "Unable to get the serving port for the application.")
+
     acc.stop_app(options.appname)
     AppScaleLogger.log("Please wait for your app to shut down.")
-    while True:
-      if acc.is_app_running(options.appname):
+
+    for _ in range(cls.MAX_RETRIES):
```
Comment (cdonati) [inline appscale/tools/appscale_tools.py]: same here

File: appscale/tools/appscale_tools.py [hunk 6]
```diff
@@ -653,23 +680,24 @@ def upload_app(cls, options):
 
     # Makes a call to the AppController to get all the stats and looks
     # through them for the http port the app can be reached on.
-    current_app = None
-    for i in range(cls.MAX_RETRIES):
+    http_port = None
+    for _ in range(cls.MAX_RETRIES):
```
Comment (cdonati) [inline appscale/tools/appscale_tools.py]: and here

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1707/

**Pull request merged**